packages/edit-site/src/index.js

```diff
@@ -84,6 +84,23 @@ export function initializeEditor( id, settings ) {
 	}
 
 	dispatch( editSiteStore ).updateSettings( settings );
+	// 🔥 Re-apply settings when style variation changes
+	if ( globalThis.wp?.data ) {
+		const { subscribe, select } = globalThis.wp.data;
+		const previousVariation = { current: null };
+
+		subscribe( () => {
+			const newVariation =
+				select( 'core/edit-site' )?.getSettings()?.styleVariation;
+
+			if ( newVariation && newVariation !== previousVariation.current ) {
+				previousVariation.current = newVariation;
+
+				// Re-dispatch settings to refresh block controls (duotone, spacing, etc.)
+				dispatch( editSiteStore ).updateSettings( settings );
+			}
+		} );
+	}
```

Comment:
There are a couple of issues with this code:

* The core code shouldn't use `window` to access methods from the packages.
* The `const previousVariation` can be just `let`. There's no need to mimic React's reference object.
* As far as I can tell, the `styleVariation` settings value doesn't exist.
* The `settings` in this case are just static settings passed from the server. Re-dispatching them shouldn't affect controls, unless there's some implicit side-effect triggered by it.